REGRESSION (10/5/2021?): [iOS] http/wpt/cross-origin-opener-policy/single-request-to-server.html is a flaky failure

cdumez opened the pull request:
#### 59340788443f8ffc2a1e09742962e074a70eb9d1
<pre>
REGRESSION (10/5/2021?): [iOS] http/wpt/cross-origin-opener-policy/single-request-to-server.html is a flaky failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=233284">https://bugs.webkit.org/show_bug.cgi?id=233284</a>
&lt;rdar://85520978&gt;

Reviewed by Geoffrey Garen.

The test was synchronously checking that the popup would be closed once receiving the broadcast message
from the popup, due to process isolation. However, because the popup code actually runs in a new process,
the broadcast message may get received before the previous process reports the popup as closed.

To address the issue, we now make sure the popup eventually closes instead.

* LayoutTests/http/wpt/cross-origin-opener-policy/single-request-to-server.html:

Canonical link: <a href="https://commits.webkit.org/252004@main">https://commits.webkit.org/252004@main</a>
</pre>

geoffreygaren reviewed (approved):
r=me

Will this test change sync back to WPT?

cdumez committed:
REGRESSION (10/5/2021?): [iOS] http/wpt/cross-origin-opener-policy/single-request-to-server.html is a flaky failure
https://bugs.webkit.org/show_bug.cgi?id=233284
<rdar://85520978>

Reviewed by Geoffrey Garen.

The test was synchronously checking that the popup would be closed once receiving the broadcast message
from the popup, due to process isolation. However, because the popup code actually runs in a new process,
the broadcast message may get received before the previous process reports the popup as closed.

To address the issue, we now make sure the popup eventually closes instead.

* LayoutTests/http/wpt/cross-origin-opener-policy/single-request-to-server.html:

Canonical link: https://commits.webkit.org/252004@main
